### ISSUE TITLE
Bluetooth controllers disconnect immediately.

### ISSUE THREAD

**Issue by TeamLinux01** [ublue-os/bazzite#456]
I installed the Deck version on my PC. System is fully up to date.
I tried pairing my 8BitDo Pro 2 and my DualShock 4 controllers inside Game mode and on the Plasma desktop.

Both controllers will pair, show that they are connected and then immediately disconnect. No inputs are registered. Re-paired them a few times, but problem persists.

These controllers worked fine on Nobara Steam Deck edition and Fedora Plasma 38. Not sure what is causing the issue.

I also have a Steam controller connected via a dongle and that works just fine. The Steam controller was not on at the time of testing.

**Comment by simon511000** [comment]
I have the same issue on my Steam Deck

**Comment by TeamLinux01** [comment]
I found out the issue.

Looking at `journalctl -u bluetooth.service` it displayed:
`Oct 24 18:56:02 console (uetoothd)[8129]: ConfigurationDirectory 'bluetooth' already exists but the mode is different. (File system: 755 ConfigurationDirectoryMode: 555)`

Changing the mode to 555 on the directory and restarting the service fixed the issue.

`sudo chmod 555 /sys/devices//pci0000:00/0000:00:14.0/usb1/1-6/1-6:1.0/bluetooth/` for me
`sudo systemctl deamon-reload && sudo systemctl restart bluetooth.service`

The controller had no problems connecting after that.

**Comment by TeamLinux01** [comment]
I can confirm that changing the line in the `bluetooth.service` in the ostree image to
```
ConfigurationDirectoryMode=0755
```
does indeed fix the issue.

**Comment by TeamLinux01** [comment]
After writing the new line to the ostree file of `bluetooth.service`, I still have an issue with initial pairing of the controller.

Without removing the pairing, if the bluetooth.service is restarted or the OS is restarted, the controller will connect and stay connected.

P.S. The touchpad on the DualShock 4 is not working on the desktop, but the rest of the controls seem to be fine.

**Comment by KyleGospo** [comment]
> P.S. The touchpad on the DualShock 4 is not working on the desktop, but the rest of the controls seem to be fine.

This is intentional. Valve ships ds-inhibit to prevent mouse functionality. You should be able to configure it to work as a mouse in Steam.

> I found out the issue.

Great find! I just pushed up a potential fix for this.

**Comment by TeamLinux01** [comment]
It might also be a bug in Bluez 5.69+ as well.
https://bbs.archlinux.org/viewtopic.php?pid=2126205#p2126205

It looks like they are having the same issue that I am experiencing.
I had this error this error be recorded when trying to pair my DS4
```
... Failed to add UUID: Failed (0x03)
... src/service.c:service_probe() input-hid profile probe failed for ...
... src/device.c:search_cb() ...: error updating services: Host is down (112)
```
Along with a displayed error Plasma's Bluetooth system settings GUI
`br-connection-profile-unavailable` and `br-connection-page-timeout`

I manually deleted the ostree file of `/etc/bluetooth/main.conf` via a live OS, also left the bluetooth.service as 0755.


After the controller is paired, it will connect without issues as soon as the bluetooth.service is restarted.

**Comment by beefchimi** [comment]
Not sure if you have seen this issue - but the advice in it helped me get a controller working via `bluetooth`: https://github.com/ublue-os/bazzite/issues/299

That being said, I am now having some other issues pairing via `bluetooth` 😢

**Comment by yodatak** [comment]
I got the same issue with Intel AX1675 bazzite steam hpc gnome 20231025

**Comment by TeamLinux01** [comment]
I just reinstalled with 2.0.2 and it has the issue. Tried pairing my 8BitDo controller in Game Mode, it connects then disconnects.

journalctl -u bluetooth.service after going back to desktop mode
```
Nov 08 17:09:15 console systemd[1]: Starting bluetooth.service - Bluetooth service...
Nov 08 17:09:15 console (uetoothd)[1166]: bluetooth.service: ConfigurationDirectory 'bluetooth' already exists but the mode is different. (File system: 755 ConfigurationDirectoryMode: 555)
Nov 08 17:09:15 console bluetoothd[1166]: Bluetooth daemon 5.70
Nov 08 17:09:15 console bluetoothd[1166]: Starting SDP server
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support csip plugin
Nov 08 17:09:15 console bluetoothd[1166]: profiles/audio/micp.c:micp_init() D-Bus experimental not enabled
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support micp plugin
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support vcp plugin
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support mcp plugin
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support bass plugin
Nov 08 17:09:15 console bluetoothd[1166]: src/plugin.c:plugin_init() System does not support bap plugin
Nov 08 17:09:15 console bluetoothd[1166]: Bluetooth management interface 1.22 initialized
Nov 08 17:09:15 console bluetoothd[1166]: src/adapter.c:reset_adv_monitors_complete() Failed to reset Adv Monitors: Failed (0x03)
Nov 08 17:09:15 console bluetoothd[1166]: Failed to clear UUIDs: Failed (0x03)
Nov 08 17:09:15 console bluetoothd[1166]: Failed to add UUID: Failed (0x03)
Nov 08 17:09:15 console bluetoothd[1166]: Failed to add UUID: Failed (0x03)
Nov 08 17:09:15 console bluetoothd[1166]: Failed to add UUID: Failed (0x03)
```

**Comment by TeamLinux01** [comment]
Huh, I was about to change some things, but when I restarted back into desktop mode, the controller connected and stayed connected.

**Comment by TeamLinux01** [comment]
I think this should be the last little bit for me to add to this issue.

I installed Nobara Deck edition and had the same exact issue. It also uses bluez 5.70.

I thought it might be the same 0550 ConfigDirectory setting in bluetooth.service, but that didn't help. Someone suggested to install blueman, but that also didn't help.
I downgraded to bluez 5.69 and it had the same issue.

`Downgraded to bluez 5.68` and the problem went away. I am convinced it is just a bug in bluez right now.

I have heard it has been patched, but a new version hasn't been pushed, so either Bazzite should consider using bluez 5.68, build bluez 5.70 with the patch (not sure where the patch is) or wait until bluez 5.71 finally comes out.

Thank you for reading, helping and understanding.
